src/chat.rs

```diff
@@ -656,7 +660,7 @@ impl Chat {
             id: self.id,
             type_: self.typ as u32,
             name: self.name.clone(),
-            archived: self.archived,
+            archived: self.visibility == ChatVisibility::Archived,
```

Comment:
didn't we want to change this to a tri-state?

Reply:
we have a tri-state.

however, i decided to leave this backward-compatible for now: the [function is experimental at all](https://github.com/deltachat/deltachat-core-rust/blob/7274197da0da22b3e33d889a9224a344d64edf0a/src/chat.rs#L648) and the whole documentation does not match what the function is doing (compare https://github.com/deltachat/deltachat-core-rust/blob/7274197da0da22b3e33d889a9224a344d64edf0a/deltachat-ffi/deltachat.h#L2701 with https://github.com/deltachat/deltachat-core-rust/blob/7274197da0da22b3e33d889a9224a344d64edf0a/src/chat.rs#L656 ) fixing that is out of scope of this pr. 

i think, when the function becomes really in use, we should think this over, also if expensive calls are needed there, naming, etc.

Reply:
ok